File: online-judge-solutions/Codeforces/1695A.py
#https://codeforces.com/problemset/problem/1695/A
#Lang		:	Python 3.8
#Time		:	62 ms
#Memory         :	0 KB
for _ in range(int(input())):
    n,m=map(int,input().split())
    z,x,y,v=-1000000000,0,0,[]
    for i in range(n):
        fila=list(map(int,input().split()))[:m]
        v.append(fila)
    for i in range(n):
        for j in range(m):
            if(v[i][j]>z):
                z=v[i][j]
                y,x=i,j
    y+=1
    x+=1
    print(max(y,n-y+1)*max(x,m-x+1))

File: online-judge-solutions/Codeforces/1695A.cpp
//https://codeforces.com/problemset/problem/1695/A
//Lang		:	GNU C++14
//Time		:	62 ms
//Memory	:	200 KB
#include<iostream>
#include<vector>
using namespace std;
int main(){
	int t,n,m;
	cin>>t;
	while(t--){
		cin>>n>>m;
		vector<vector<int> > v(n,vector<int>(m));
		int z=-1000000000,x=0,y=0;
		for(int i=0;i<n;i++){
			for(int j=0;j<m;j++){
				cin>>v[i][j];
				if(v[i][j]>z){
					z=v[i][j];
					y=i,x=j;
				}
			}
		}
		y++;x++;
		cout<<max(y,n-y+1)*max(x,m-x+1)<<endl;
	}
	return 0;
}
